### PR TITLE
CBO-299 LGFS grad fee calc

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
LGFS graduated fee calculation

#### Ticket

[CBO-299](https://dsdmoj.atlassian.net/browse/CBO-299)

#### Why
So users do not have to calculate amounts to be claimed using spreadsheets

#### How
Add  a new subclass of `Claims::FeeCalculator::Calculate`
and namespaced JS module to handle the graduated fee pricing
logic (where multiple units are applicable and calling a different
API endpoint) and separate concerns.

--------

#### TODO (wip)
 - [x] spike - get it working - Approved
 - [x] unit tests
 - [x] feature tests
 - [x] refactor js - sub-modules
 - [x] refactor controller
 - [x] persisting whether calculated or not `price_calculated?` attribute

### CODE REVIEW

File: app/services/claims/fee_calculator/calculate.rb [hunk 1]
```diff
@@ -49,30 +49,34 @@ def setup(options)
...
-          # example, actual number of defendants/cases. This is because we should based payments
-          # on what is asked for.
-          # options[:number_of_defendants] = 1
-          # options[:number_of_cases] = 1
-        end
-      end
+        raise 'implement in subclasses'
+      end
+
+      # def amount
```
Comment (colinbruce) [inline app/services/claims/fee_calculator/calculate.rb]: Does the commented out part need to stay?

File: features/page_objects/claim_form_page.rb [hunk 2]
```diff
@@ -22,7 +22,6 @@ class ClaimFormPage < SitePrism::Page
   element :court, "#s2id_autogen1"
   element :case_type, "#s2id_autogen2"
   element :case_number, "#claim_case_number"
-  # element :claim_advocate_category_junior_alone, "#claim_advocate_category_junior_alone" # TODO: remove if no longer needed
```
Comment (colinbruce) [inline features/page_objects/claim_form_page.rb:25]: Can be removed?

**Review by colinbruce — APPROVED** [review]
Looking good 👍 
Couple of minor non-blocking comments on commented out sections

File: app/services/claims/fee_calculator/calculate.rb (hunk 1 above)
Comment (colinbruce) [reply app/services/claims/fee_calculator/calculate.rb]: Sorry, due to the PR size, I reviewed each commit in sequence

File: features/page_objects/claim_form_page.rb (hunk 2 above)
Comment (jsugarman) [reply features/page_objects/claim_form_page.rb:25]: yep 👍

Comment (jsugarman) [reply features/page_objects/claim_form_page.rb:25]: actually has been in later commit but now showing for diff reasons i guess

**Pull request merged**